commit ba1feec0e1541ec7fd98d99b31d02c4d96abd129 (refs/changes/97/497/5)
Author: Ryan Zezeski <rpz@joyent.com>
Date:   2016-09-22T10:41:16-04:00 (3 years, 1 month ago)
    
    OS-5330 zoneadm mounting an lx or joyent branded zone fails
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/overlay/generic/etc/zones/SUNWdefault.xml b/overlay/generic/etc/zones/SUNWdefault.xml
index cfe27a0f..070412f6 100644
--- a/overlay/generic/etc/zones/SUNWdefault.xml
+++ b/overlay/generic/etc/zones/SUNWdefault.xml
@@ -28,5 +28,5 @@
 
 <!DOCTYPE zone PUBLIC "-//Sun Microsystems Inc//DTD Zones//EN" "file:///usr/share/lib/xml/dtd/zonecfg.dtd.1">
 
-<zone name="default" zonepath="" autoboot="false" brand="liveimg">
+<zone name="default" zonepath="" autoboot="false" brand="joyent">
 </zone>
diff --git a/overlay/generic/lib/svc/method/fs-joyent b/overlay/generic/lib/svc/method/fs-joyent
index 69729346..e0ce91c4 100755
--- a/overlay/generic/lib/svc/method/fs-joyent
+++ b/overlay/generic/lib/svc/method/fs-joyent
@@ -121,10 +121,19 @@ if [ $? -ne 0 ]; then
     # For the system zpool, mount and configure all system datasets
     zpool status -v ${SYS_ZPOOL}
     if [ $? -eq 0 ]; then
+        # Stash the SUNWdefault.xml file so we can update the
+        # persistent version after mounting zones/config.
+        cp /etc/zones/SUNWdefault.xml /tmp/
+
         mount_zfs ${SYS_ZPOOL}/var /var
         mount_zfs ${SYS_ZPOOL}/config /etc/zones
         mount_zfs ${SYS_ZPOOL}/opt /opt
 
+        # Update the the persistent SUNWdefault.xml file to match the
+        # contents on ramdisk now that zones/config is mounted.
+        cp /tmp/SUNWdefault.xml /etc/zones/
+        rm -f /tmp/SUNWdefault.xml
+
         if [[ -z $(/bin/bootparams | grep '^smartos=true') ]]; then
             mkdir -p /opt/smartdc/agents/smf
             mount -O -F lofs /var/svc/manifest/site /opt/smartdc/agents/smf
